quickwit-metastore/src/metastore_resolver.rs

```diff
@@ -126,11 +126,14 @@ impl MetastoreUriResolver {
     }
 
     /// Resolves the given URI.
-    pub async fn resolve(&self, uri: &str) -> Result<Arc<dyn Metastore>, MetastoreResolverError> {
-        let protocol = uri.split("://").next().ok_or_else(|| {
+    pub async fn resolve<S: AsRef<str>>(
```

Comment:
Adding `S: AsRef<str>` allowed me to not change `.resolve(&metastore_uri)` to `.resolve(metastore_uri.as_ref())` all over the place. We can get rid of it once we change the signature of `resolve(&str)` to `resolve(&Uri)`